From 55dffd98ef941e2e8dfd9db7dc0034827e65b3db Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Sun, 12 Jun 2022 22:29:41 +0530
Subject: [PATCH] ath12k: Print htt_peer_stats only when enabled

htt_peer_stats are displayed without enabling htt_stats_type 11. Fix
them by adding necessary checks when trying to open peer stats without
issuing stats type 11.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c | 7 ++++---
 drivers/net/wireless/ath/ath12k/debugfs_sta.c       | 6 ++++++
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
index 2fcead6..b5452de 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
@@ -7075,8 +7075,7 @@ static ssize_t ath12k_write_htt_stats_type(struct file *file,
 	if (type >= ATH12K_DBG_HTT_NUM_EXT_STATS)
 		return -E2BIG;
 
-	if (type == ATH12K_DBG_HTT_EXT_STATS_RESET ||
-	    type == ATH12K_DBG_HTT_EXT_STATS_PEER_INFO)
+	if (type == ATH12K_DBG_HTT_EXT_STATS_RESET)
 		return -EPERM;
 
 	ar->debug.htt_stats.type = type;
@@ -7203,7 +7202,9 @@ static int ath12k_open_htt_stats(struct inode *inode, struct file *file)
 	u8 type = ar->debug.htt_stats.type;
 	int ret;
 
-	if (type == ATH12K_DBG_HTT_EXT_STATS_RESET)
+	if (type == ATH12K_DBG_HTT_EXT_STATS_RESET ||
+	    type == ATH12K_DBG_HTT_EXT_STATS_PEER_INFO ||
+	    type == ATH12K_DBG_HTT_EXT_PEER_CTRL_PATH_TXRX_STATS)
 		return -EPERM;
 
 	mutex_lock(&ar->conf_mutex);
diff --git a/drivers/net/wireless/ath/ath12k/debugfs_sta.c b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
index f04ce82..02c99f7 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_sta.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
@@ -535,8 +535,14 @@ ath12k_dbg_sta_open_htt_peer_stats(struct inode *inode, struct file *file)
 	struct ath12k_sta *arsta = (struct ath12k_sta *)sta->drv_priv;
 	struct ath12k *ar = arsta->arvif->ar;
 	struct debug_htt_stats_req *stats_req;
+	int type = ar->debug.htt_stats.type;
 	int ret;
 
+	if ((type != ATH12K_DBG_HTT_EXT_STATS_PEER_INFO &&
+	     type != ATH12K_DBG_HTT_EXT_PEER_CTRL_PATH_TXRX_STATS) ||
+	    type == ATH12K_DBG_HTT_EXT_STATS_RESET)
+		return -EPERM;
+
 	stats_req = vzalloc(sizeof(*stats_req) + ATH12K_HTT_STATS_BUF_SIZE);
 	if (!stats_req)
 		return -ENOMEM;
-- 
2.7.4

